go/vt/vttablet/onlineddl/executor.go

```diff
@@ -3246,17 +3265,23 @@ func (e *Executor) reviewRunningMigrations(ctx context.Context) (countRunnning i
 				// understand whether "now is a good time" or "not there yet"
 				_ = e.updateMigrationReadyToComplete(ctx, uuid, isReady)
 				if !isReady {
-					return nil
-				}
-				if postponeCompletion {
-					// override. Even if migration is ready, we do not complete it.
+					// The migration is not ready yet.
 					return nil
 				}
 				if strategySetting.IsInOrderCompletion() {
+					var pendingMigrationsCount uint64
 					if len(pendingMigrationsUUIDs) > 0 && pendingMigrationsUUIDs[0] != onlineDDL.UUID {
-						// wait for earlier pending migrations to complete
-						return nil
+						pendingMigrationsCount = getInOrderCompletionPendingCount(onlineDDL, pendingMigrationsUUIDs)
+						postponeCompletion = true
```

Comment:
This `if` statement is corrct, but also confusing. We may as well just

```go
pendingMigrationsCount := getInOrderCompletionPendingCount(onlineDDL, pendingMigrationsUUIDs)
```

The result will be the same. Iterating an empty array, or iterating an array when our UUID is first, is not wasting any more resources than this `if` statement, and I believe the code is simpler and more readable this way.